programs/cryptid/src/instructions/propose_transaction.rs

```diff
@@ -116,14 +116,30 @@ pub fn propose_transaction<'a, 'b, 'c, 'info>(
     // Account indexes must reflect this, so the first entry
     // in the remaining accounts is referred to in the abbreviated instruction data as index 4,
     // despite being index 0 in the remaining accounts.
-    // TODO validate that the account indices are all valid, given the above i.e. that no index exceeds remaining_accounts.length + 4
     ctx.accounts.transaction_account.accounts = all_accounts.iter().map(|a| *a.key).collect();
 
+    // TODO: Set slot OR move any Slot / Expiry constraints to middleware
+    // ctx.accounts.transaction_account.slot = Clock::get()?.slot;
     ctx.accounts.transaction_account.did = *ctx.accounts.did.key;
     ctx.accounts.transaction_account.instructions = instructions;
     ctx.accounts.transaction_account.cryptid_account = *ctx.accounts.cryptid_account.key;
     ctx.accounts.transaction_account.approved_middleware = None;
 
+    // Make sure that all instructions reference accounts in bound.
+    let max_account_reference = ctx
+        .accounts
+        .transaction_account
+        .instructions
+        .iter()
+        .map(|x| x.get_max_account_index())
+        .max()
+        .unwrap_or(0);
```

Comment:
I was really impressed that the compiler managed to produce identical assembly code!

https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=6b0b0472f07888297377c6953667cb0c

I'll leave this as a slightly more brief alternative if you're interested.

```rust
    let max_account_reference = ctx
        .accounts
        .transaction_account
        .instructions
        .iter()
        .fold(0, |acc, &x| {acc.max(x.get_max_account_index())});
```

Reply:
I take it, thanks!

Reply:
From copy_paste I get
```
134 |         .fold(0, |acc, &x| {acc.max(x.get_max_account_index())});
    |                        ^-
    |                        ||
    |                        |data moved here
    |                        |move occurs because `x` has type `abbreviated_instruction_data::AbbreviatedInstructionData`, which does not implement the `Copy` trait
    |                        help: consider removing the `&`: `x`
```
So I updated it to
```
.fold(0, |acc, x|{acc.max(x.get_max_account_index())});
```